syntax/go.vim

```diff
@@ -452,4 +465,5 @@ syn sync minlines=500
 
 let b:current_syntax = "go"
 
-" vim: sw=2 ts=2 et
```

Comment:
I think we want to keep the modeline; you you add this back please?

Reply:
Oh, sry I guess I removed modeline within my tests down there and didn't realize, I was also trying to create a FoldText() function so in the cases we fold the declarations the text would contain the subject (func/type/const/var),

I made the suggested changes